### ISSUE TITLE
Bot isn't transfering based on moves set

### ISSUE THREAD

**Issue by jasio77** [NECROBOTIO/NecroBot#4540]
### Expected Behavior

- Transfer pokemon based on moves set.

### Actual Behavior

- It doesn't transfer them, like it wasn't set.

### Your config.json

http://pastebin.com/aNdxGRmZ

### Steps to Reproduce

- Add moves to pokemon in PokemonTransferFilter.

### Other Information
Version: 0.8.5
Windows Version: Windows 10 Home, build 1511, x64
What is affected (catch, transfer, evolve, recycle, snipe, powerup/levelup, other): Transfer

**Comment by nfettinger** [comment]
Have you talked to helpers in our Discord chat rooms yet to ensure this is not a configuration issue?

**Comment by ruchern** [comment]
Check the operator for Moves.

If you wish to strictly transfer any that is not of the moves combo, then you have to use yes. Otherwise so long as one move appears in the move list, it will not be transferred.

**Comment by fuckNIANTIC** [comment]
fuck NIANTIC

**Comment by spamfighter82** [comment]
> If you wish to strictly transfer any that is not of the moves combo, then you have to use yes. Otherwise so long as one move appears in the move list, it will not be transferred.

Thats not true

```
    "Dragonite": {
      "KeepMinCp": 1,
      "KeepMinLvl": 20,
      "UseKeepMinLvl": true,
      "KeepMinIvPercentage": 90.0,
      "KeepMinDuplicatePokemon": 0,
      "Moves": [
        "dragonBreathFast",
        "dragonClaw"
      ],
      "KeepMinOperator": "and",
      "MovesOperator": "and"
    }
```

keeps

Dragonite	22	2179	92.3350958536851	122	122	DragonBreathFast	HyperBeam

**Comment by Usaklioglan** [comment]
"KeepMinDuplicatePokemon": 0 , must be 1 ?! otherwise it will not keep it if you have more than 1 dragonite

**Comment by spamfighter82** [comment]
Nah, KeepMinDuplicatePokemon = 1 would keep crap Pknm, even with 0 it will not get transferred but it should if Moves actually would work.

**Comment by arxnoa** [comment]
I'm understanding that as 
"KeepMinDuplicatePokemon": 0  =  "do not keep any dragonite"

If set to 1, wouldn't it keep 1 dragonite that matches keepmin level/iv and moves

**Comment by spamfighter82** [comment]
KeepMinDuplicatePokemon" is not the issue here

**Comment by jasio77** [comment]
I set it to keep Snorlax with Zen Headbutt and Body Slam. Instead bot keeps Snorlax with Lick and Earthquake. Moves should be the last filter, because Snorlax can have 100 IV but weak moves. I have KeepMinDuplicatePokemon 0, because 1 keeps 1 duplicate = 2 Pokémons.

**Comment by jasio77** [comment]
And yes, I talked with other people and they have the same issue.

**Comment by EndTrophy** [comment]
It might be, that the bot does not recognize 
dragonBreath**_Fast_**
could you try just
dragonBreath
?

**Comment by jasio77** [comment]
@EndTrophy No, when I run bot without Fast it gives me exceptions and crashes. Fast is for fast move, not charge move. The same is in dumps, it shows fast attack (not charge) as DragonBreathFast.

**Comment by spamfighter82** [comment]
nope thats the name of the move. 
at the moment you cant filter for CP and IV and 2 moves, which is was most ppl want.

**Comment by jasio77** [comment]
But, I don't want that. I set MovesOperator to "or", so I can have pokemon with minimum 1 best move. If there's option to set MovesOperator, I want it to work as I want, not what most people want.

**Comment by spamfighter82** [comment]
well and if i set it to AND i want both moves, doesnt work either way

**Comment by ruchern** [comment]
>DragonBreathFast

The move name is actually "dragonBreath" in JSON.

Pleases correct it and try again.

**Comment by spamfighter82** [comment]
That transferred my best Dragonite which accutally had >90IV >lvl20 and the right moveset, thanks mate

**Comment by jasio77** [comment]
@ruchern I don't know why, but dragonBreath and dragonBreathFast don't give exception. But when i try to delete "Fast" in zenHeadbuttFast (so it's zenHeadbutt) it gives exception and crashes. Why some moves require "Fast" and some are working fine without it. (I'm talking about fast moves, I know that charge moves are without "Charge" or anything :P)

**Comment by jasio77** [comment]
So after @spamfighter82 I won't let bot do something more that proper login-in and displaying 10 best Pokemons in inventory. (with dragonBreathFast changed to dragonBreath)

**Comment by spamfighter82** [comment]
DragonBreath and DragonBreahFast are both in the json, dont asked my why.
anyway trying this had cost me a 98IV lvl 28.5 dragonBreath/dragonClaw Dragonite, im dont with this shit today, dont really have anymore pkmn to try

**Comment by GX-Q7** [comment]
Filtering out pokemon for transfer based on moves needs to be done before regular filtering based on cp/iv.

**Comment by jasio77** [comment]
Operator "or" should be fixed too. Because "or" & "and" doesn't do anything.

**Comment by jasio77** [comment]
@nfettinger Operator "or" should be fixed too. Because "or" & "and" doesn't do anything. Can you add label "bug"? It has been reported several times and hasn't been fixed.
